cpp/src/join/mixed_join_semi.cu

```diff
@@ -107,12 +106,7 @@ std::unique_ptr<rmm::device_uvector<size_type>> mixed_join_semi(
 
   auto const right_num_rows{right_conditional.num_rows()};
   auto const left_num_rows{left_conditional.num_rows()};
-  auto const swap_tables = (join_type == join_kind::INNER_JOIN) && (right_num_rows > left_num_rows);
-
-  // The "outer" table is the larger of the two tables. The kernels are
-  // launched with one thread per row of the outer table, which also means that
-  // it is the probe table for the hash
-  auto const outer_num_rows{swap_tables ? right_num_rows : left_num_rows};
```

Comment:
We probably shouldn't remove the swap table logic. For inner join, we want to always build the hash map from the smaller table since table insertions are much more expensive than lookups.

Reply:
I was somewhat confused here. I see that [at the function start](https://github.com/tgujar/cudf/blob/abdca82e7f6d1a7386930a2e0d30f987b2f6a633/cpp/src/join/mixed_join_semi.cu#L99) we have an explicit check that this isn't an inner join
```
CUDF_EXPECTS((join_type != join_kind::INNER_JOIN) && (join_type != join_kind::LEFT_JOIN) &&
                 (join_type != join_kind::FULL_JOIN),
               "Inner, left, and full joins should use mixed_join.");
```

Reply:
Good catch. It's probably some leftover when copying the same code around for different join types. Nice cleanups!